spec/query.spec.ts

```diff
@@ -232,13 +245,79 @@ describe("APIQuery", () => {
     new APIQuery(bundle);
     // Passing is not raising an exception
   });
+
+  it("throw eror on invalid phase value", () => {
+    expect(() => {
+      new APIQuery({
+        resourceType: "Bundle",
+        type: "collection",
+        entry: [
+          {
+            resource: {
+              resourceType: "Parameters",
+              parameter: [
+                {
+                    "name": "phase",
+                    "valueString": "phase-23"
+                }
+              ],
+            },
+          },
+        ],
+      }).toString()
+    }).toThrowError("Invalid value of phase. Permissible values are |early-phase-1|,|phase-0|,|phase-1|,|phase-2|,|phase-3|,|phase-4|");
```

Comment:
Should these be checking the exact error message? Is it possible for the permissible types to change later?